### PR TITLE
Only send Amplitude analytics events when connected to an MT US environment

### PR DESCRIPTION
**Pull request by chdorner-snyk**
### Description

Disables sending analytics events to Amplitude when connected to a non-US MT or FedRAMP environment.

Co-authored-by: Knut <knut.funkel@snyk.io>

https://snyksec.atlassian.net/browse/IDE-24

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] ~~README.md updated, if user-facing~~

### CODE REVIEW

File: Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs [hunk 1]
```diff
@@ -111,6 +112,18 @@ public bool IsFedramp()
...
 
+        /// <summary>
+        /// Checks if the current endpoint permits sending external analytics events
+        /// </summary>
+        /// <returns></returns>
+        public bool IsAnalyticsPermitted()
+        {
+            var endpointUri = new Uri(this.GetAppCustomEndpoint());
+
+            string[] permittedHosts = ["app.snyk.io", "app.us.snyk.io"];
```
Comment (bastiandoetsch) [inline Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs:123]: Wondering - what if no custom endpoint is specified or snyk.io/api is specified?

File: Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs [hunk 2]
```diff
@@ -388,11 +401,11 @@ private string GetAppCustomEndpoint()
             var endpoint = this.customEndpoint.IsNullOrEmpty() ? "https://app.snyk.io" : this.customEndpoint.RemoveTrailingSlashes();
```
Comment (bastiandoetsch) [inline Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs:401]: This is missing the path (`/api/).

Generally, we need either `api.<tenant>.snyk[gov].io` or `app.<tenant>.snyk[gov].io/api`.

File: Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs [hunk 3]
```diff
@@ -388,11 +401,11 @@ private string GetAppCustomEndpoint()
...
             Uri uri = new Uri(endpoint);
 
-            if (!uri.Host.StartsWith("app") && uri.Host.EndsWith("snyk.io"))
+            if (!uri.Host.StartsWith("app") && (uri.Host.EndsWith("snyk.io") || uri.Host.EndsWith("snykgov.io")))
             {
                 return endpoint.Replace("https://", "https://app.").RemoveFromEnd("api");
             }
-            else if (uri.Host.StartsWith("app") && uri.Host.EndsWith("snyk.io"))
+            else if (uri.Host.StartsWith("app") && (uri.Host.EndsWith("snyk.io") || uri.Host.EndsWith("snykgov.io")))
```
Comment (bastiandoetsch) [inline Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs:408]: This whole method looks really wrong to me - can we have some (additional) tests for this?

Comment (bastiandoetsch) [reply Snyk.VisualStudio.Extension.Shared/Settings/SnykGeneralOptionsDialogPage.cs:408]: Maybe it's just wrongly named (-> getAppHostFromEndpoint) and not used to pass to the CLI. If it is passed to the CLI in this form, I'm pretty sure that's why authenticate for example does not work reliably (we had bug reports on this).

**Review by acke — APPROVED** [review]
Very nice refactor. Really like the use of parametrized tests!

**Pull request merged**